starfish/image/_single.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import collections
+from typing import Tuple
+
+from slicedimage import ImageFormat
+from slicedimage.io import resolve_url
+
+from ._base import ImageBase
+
+
+class SingleImage(ImageBase):
+    def __init__(self, data):
+        self._data = data
+
+    @classmethod
+    def from_url(cls, relativeurl: str, baseurl: str):
+        backend, name, baseurl = resolve_url(relativeurl, baseurl)
+        extension = os.path.splitext(relativeurl)[1].lstrip(".")
+        format = ImageFormat.find_by_extension(extension)
+        with backend.read_file_handle(name) as fh:
+            return SingleImage(format.reader_func(fh))
+
+    @property
+    def numpy_array(self):
+        data = self._data.view()
+        data.setflags(write=False)
+        return data
+
+    @numpy_array.setter
+    def numpy_array(self, data):
+        self._data = data.view()
+        data.setflags(write=False)
+
+    @property
+    def raw_shape(self) -> Tuple[int]:
+        return self._data.shape
+
+    @property
+    def shape(self) -> collections.OrderedDict:
+        return collections.OrderedDict()
+
+    def write(self, filepath):
```

Comment:
Here and elsewhere (maybe new PR?) I would either change this to `write_npy` or (better) add a parameter that allows the user to dictate the output type and make `.tif` the default. My expectation for a write command is to write the same format that I read (which in this case, is `.tif`, since that's what we read.

Reply:
@dganguli has told me in the past that TIF -> TIF does not work for starfish because the intermediate results are floating point.  is this no longer the case?